joblib/numpy_pickle.py

```diff
@@ -679,7 +710,7 @@ def load(filename, mmap_mode=None):
         fobj = filename
         filename = getattr(fobj, "name", "")
         with _validate_fileobject_and_memmap(fobj, filename, mmap_mode) as (fobj, _):
-            obj = _unpickle(fobj)
+            obj = _unpickle(fobj, ensure_native_byte_order=True)
```

Comment:
The docstring of `joblib.load` should be updated to document this magic.

I think we should also expose `ensure_native_byte_order="auto"` in joblib load to:

- passing `False` makes it possible to disable this feature for users we do not want the automated conversion,
- "auto" means `True` when `mmap_mode=None` else `False` otherwise.
- passing `True` and `mmap_mode != None` would raise `ValueError` with a message similar to what was suggested in https://github.com/joblib/joblib/pull/1561/files#r1553524980.